base/templates/service.yaml

```diff
@@ -14,7 +14,13 @@ spec:
   ports:
     - port: {{ .Values.service.srcPort }}
       targetPort: {{ .Values.service.dstPort }}
-      protocol: TCP
-      name: {{ .Values.service.name | default "http" }}
+      protocol: {{ default "TCP" .Values.service.protocol }}
+      name: {{ .Values.service.name | default "http" }}    
+    {{- range $additionalPortIndex, $additionalPort := .Values.service.additionalPorts }}
+    - port: {{ $additionalPort.srcPort }}
+      targetPort: {{ $additionalPort.dstPort }}
+      protocol: {{ default "TCP" $additionalPort.protocol }}
+      name: {{ default "http" $additionalPort.name }}    
```

Comment:
you can't use same `name` for this. consider to change default name to others than http

Reply:
oke kang ade.
any suggestions?  

how about `https` or grpc ?

Reply:
i think don't use the standard one, try something like `port-name` or anything that isn't standard name.

Reply:
sama make sure juga kasi comment "jangan ngasi nama yang sama untuk `name`" di values.yaml biar orang2 bisa aware,

Reply:
FYI kalo namanya sama bakal error soalnya. jadi ini cukup penting hhe

Reply:
ahhh I see.
how about, if we make this variable become required, not use the default.
because this is an array and will be looping,

so if the engineer does not add the value the print result will same name if he creates more than 2 port

Reply:
yap yap bener kang ade
baru keingetan.

better aku bikin required aja kalau gtu.
jadi biar engineer aware itu wajib diisi

Reply:
please check again kang @ademahmudf 

now, if the engineer adds values like this :
```
service:
  additionalPorts: 
    - srcPort: 443
      dstPort: 443
      protocol: TCP #default TCP
      name: https #default http
    - srcPort: 10001
      dstPort: 10001
```
the result will become like this
![image](https://user-images.githubusercontent.com/37339379/159464297-f82358d9-e7ff-4e7b-8889-a42d4eb15031.png)